// Wonto/Create(3-3).js
var weapon = require('origindb')('weapon');
var acc = require('origindb')('acc');
var armor = require('origindb')('armor');
var sect = require('origindb')('sect');
var monsters = require('origindb')('monsters');
var constant = 1/49;

module.exports = {

	makehp: function(lvl, w, ar, ac, cls){
		var classHp = sect(cls).get('hp');
		var wep = weapon(w).get('hp'); var arm = armor(ar).get('hp'); var access = acc(ac).get('hp');
		var hp = Math.floor((((lvl*constant) - constant)*(classHp[0] - classHp[1]) + classHp[1]) + (wep + arm + access)*((lvl*constant) - constant));
		return hp;
	},
	makepdef: function(lvl, w, ar, ac, cls){
		var arm = armor(ar).get('pdef'); var access = acc(ac).get('pdef');
		var pdef = Math.floor((arm + access)*((lvl*constant)-constant));
		return pdef;
	},
	makemdef: function(lvl, w, ar, ac, cls){
		var arm = armor(ar).get('mdef'); var access = acc(ac).get('mdef');
		var mdef = Math.floor((arm + access)*((lvl*constant)-constant));
		return mdef;
	},
	makepow: function(lvl, w, ar, ac, cls){
		var classPow = sect(cls).get('pow');
		var wep = weapon(w).get('pow'); var access = acc(ac).get('pow');
		var pow = Math.floor((((lvl*constant) - constant)*(classPow[0] - classPow[1]) + classPow[1]) + (wep + access)*((lvl*constant) - constant));
		return pow;
	},
	makedex: function(lvl, w, ar, ac, cls){
		var classDex = sect(cls).get('dex');
		var wep = weapon(w).get('dex'); var access = acc(ac).get('dex');
		var pow = Math.floor((((lvl*constant) - constant)*(classDex[0] - classDex[1]) + classDex[1]) + (wep + access)*((lvl*constant) - constant));
		return pow;
	},
	makemag: function(lvl, w, ar, ac, cls){
		var classMag = sect(cls).get('mag');
		var wep = weapon(w).get('mag'); var access = acc(ac).get('mag');
		var pow = Math.floor((((lvl*constant) - constant)*(classMag[0] - classMag[1]) + classMag[1]) + (wep + access)*((lvl*constant) - constant));
		return pow;
	},
	makespe: function(lvl, w, ar, ac, cls){
		var wep = weapon(w).get('spe'); var arm = armor(ar).get('spe'); var access = acc(ac).get('spe');
		var spe = 10 + wep + arm + access
		return spe;
	},
	monhp: function(lvl, mon){
		var monHp = mon['hp'];
		var hp = Math.floor((lvl*constant - constant)*(monHp[0] - monHp[1]) + monHp[1])
		return hp;
	},
	monpdef: function(lvl, mon){
		var monPdef = mon['pdef'];
		var pdef = Math.floor((lvl*constant - constant)*(monPdef[0] - monPdef[1]) + monPdef[1])
		return pdef;
	},
	monmdef: function(lvl, mon){
		var monMdef = mon['mdef'];
		var mdef = Math.floor((lvl*constant - constant)*(monMdef[0] - monMdef[1]) + monMdef[1])
		return mdef;
	}
}
