// Wonto/dCreate.js
var players = require('origindb')('players');
var weapon = require('origindb')('weapon');
var sect = require('origindb')('sect');

module.exports = {

	makehp: function(lvl, wep, cls){
		var hp = 10 + Math.round((lvl*1.5) + ((sect(toId(cls)).object().hp + weapon(toId(wep)).object().hp)*((1 + (7/100)*lvl))));
		return hp;
	},
	makepdef: function(lvl, wep, cls){
		var pdef = Math.round((sect(toId(cls)).object().pdef + weapon(toId(wep)).object().pdef)*(((lvl*13)/140)+1));
		return pdef;
	},
	makemdef: function(lvl, wep, cls){
		var mdef = Math.round((sect(toId(cls)).object().mdef + weapon(toId(wep)).object().mdef)*(((lvl*13)/140)+1));
		return mdef;
	},
	makepow: function(lvl, wep, cls){
		var pow = Math.round((sect(toId(cls)).object().pow + weapon(toId(wep)).object().pow)*((lvl/10)+1));
		return pow;
	},
	makedex: function(lvl, wep, cls){
		var dex = Math.round((sect(toId(cls)).object().dex + weapon(toId(wep)).object().dex)*((lvl/10)+1));
		return dex;
	},
	makemag: function(lvl, wep, cls){
		var mag = Math.round((sect(toId(cls)).object().mag + weapon(toId(wep)).object().mag)*((lvl/10)+1));
		return mag;
	},
	makemov: function(lvl, wep, cls){
		var mov = (sect(toId(cls)).object().mov + weapon(toId(wep)).object().mov);
		return mov;
	},
	makespe: function(lvl, wep, cls){
		var spe = (sect(toId(cls)).object().spe + weapon(toId(wep)).object().spe);
		return spe;
	},
	price: function(lvl){
		var result = Number(lvl)*5;
		return result;
	},
}
